configure.ac

```diff
@@ -818,27 +818,25 @@ AC_ARG_ENABLE(esi,
                   Enabling ESI will cause squid reverse proxies to be capable
                   of the Edge Acceleration Specification (www.esi.org).]),
                  [squid_opt_use_esi=$enableval],[])
-HAVE_LIBEXPAT=0
-EXPATLIB=
 HAVE_LIBXML2=0
 XMLLIB=
 
 # ESI support libraries: expat
+AH_TEMPLATE(HAVE_LIBEXPAT,[Define to 1 if you have the expat library])
 SQUID_AUTO_LIB(expat,[ESI expat library],[LIBEXPAT])
 AS_IF([test "x$squid_opt_use_esi" != "xno" -a "x$with_expat" != "xno"],[
-  AC_CHECK_LIB([expat], [main], [EXPATLIB="-lexpat"; HAVE_LIBEXPAT=1])
-  AC_CHECK_HEADERS([expat.h])
-  AC_DEFINE_UNQUOTED(HAVE_LIBEXPAT, $HAVE_LIBEXPAT, [Define to 1 if you have the expat library])
-  AS_IF([test "x$HAVE_LIBEXPAT" = "x1"],[
-    squid_opt_use_esi=yes
-  ],[
-    AS_IF([test "x$with_expat" = "xyes"],[
-      AC_MSG_ERROR([Required library expat not found.])
-    ],[
-      AC_MSG_NOTICE([Library expat not found.])
-    ])
-  ])
+  AC_CHECK_LIB([expat],[main],[
+    LIBEXPAT_LIBS="-lexpat"
+    AC_CHECK_HEADERS([expat.h])
+    AC_DEFINE(HAVE_LIBEXPAT,1,[Define to 1 if you have the expat library])
+  ])
+  AS_IF([test "x$LIBEXPAT_LIBS" != "x"],[squid_opt_use_esi=yes],
+    [test "x$with_expat" = "xyes"],[AC_MSG_ERROR([Required library expat not found.])],
+    [AC_MSG_NOTICE([Library expat not found.])]
+  )
 ])
+AM_CONDITIONAL(ENABLE_LIBEXPAT,true)
```

Comment:
The always-true condition for enabling libexpat does not look right. Please test where libexpat should _not_ be enabled.

Reply:
Thanks. Fixed in [1728cd6](https://github.com/squid-cache/squid/pull/1143/commits/1728cd6dd4dc655ba0a0c11eb36ffc09670b9fcc)